hazelcast/src/main/java/com/hazelcast/config/CompactSerializationConfig.java

```diff
@@ -23,110 +23,159 @@
 
 import static com.hazelcast.internal.util.Preconditions.checkNotNull;
 
 /**
- * In 5.0, this feature is disabled by default and has to be enabled via configuration
+ * In 5.0, this feature is disabled by default and has to be enabled via
+ * configuration
  * <p>
- * Once enabled the classes which does not match to any other serialization methods will be serialized in
- * the new format automatically via help of reflection without needing any other configuration.
- * For this case, the class should not implement `Serializable`, `Externalizable` , `Portable` ,
- * `IdentifiedDataSerializable`, `DataSerializable` or the class should not be registered as a custom serializer.
+ * Once enabled the classes which does not match to any other serialization
+ * methods will be serialized in the new format automatically via help of
+ * reflection without needing any other configuration. For this case, the class
+ * should not implement `Serializable`, `Externalizable`, `Portable`,
+ * `IdentifiedDataSerializable`, `DataSerializable` or the class should not be
+ * registered as a custom serializer.
  * <p>
- * Automatic serialization via reflection can de/serialize classes having an accessible empty constructor only.
- * Only types in {@link CompactWriter}/{@link CompactReader} interface are supported as fields.
- * For any other class as the field type, it will work recursively and try to de/serialize a sub-class.
- * Thus, if any sub-fields does not have an accessible empty constructor, deserialization fails with
- * HazelcastSerializationException.
+ * For automatic serialization with reflection, only types in
+ * {@link CompactWriter}/{@link CompactReader} interface and {@code char},
+ * {@code Character}, {@code enum}, and their arrays are supported as fields.
+ * For any other field type, the reflective serializer will work recursively and
+ * try to de/serialize those as nested Compact objects.
  * <p>
- * To explicitly configure a class to be serialized via Compact format following methods can be used.
- * {@link #register(Class)}
- * {@link #register(Class, String, CompactSerializer)} }
+ * To explicitly configure a class to be serialized with Compact format
+ * following methods can be used.
+ * <ul>
+ *     <li>{@link #addSerializer(CompactSerializer)}</li>
+ *     <li>{@link #setSerializers(CompactSerializer[])}</li>
+ *     <li>{@link #addClass(Class)}</li>
+ *     <li>{@link #setClasses(Class[])}</li>
+ * </ul>
  * <p>
- * On the last two methods listed above reflection is not utilized instead given serializer is used to
- * serialize/deserialize users objects.
+ * On the first two methods listed above, reflection is not utilized. Instead,
+ * the given serializer is used to serialize/deserialize users objects.
  *
- * @since Hazelcast 5.0 as BETA. The final version will not be backward compatible with the Beta.
- * Do not use BETA version in production
+ * @since Hazelcast 5.0 as BETA. The final version will not be backward
```

Comment:
Will we change this since tag everywhere with :
```suggestion
 * @since Hazelcast 5.2
```

Reply:
Yeah, I was waiting for all the PRs to be merged. There are lots of places where we have such beta tags and since values